tmk_core/protocol/chibios/usb_main.c

```diff
@@ -606,6 +621,14 @@ static void set_led_transfer_cb(USBDriver *usbp) {
     }
 }
 
+#ifdef LAMPARRAY_ENABLE
+static universal_lamparray_response_t universal_lamparray_report_buf;
+
+static void set_lamparray_transfer_cb(USBDriver *usbp) {
+    lamparray_handle_set(&universal_lamparray_report_buf);
+}
```

Comment:
This callback gets called at interrupt time too, and therefore imposes severe restrictions on the called functions.  The current call tree seems to be _mostly_ safe (the involved `rgb_matrix_overlay_*()` functions modify only some memory buffers, and the pointer manipulations in there might be safe, because that's the only code that writes into those pointers), but the fact that those functions need to be interrupt-safe should probably be noted in comments.